### PR TITLE
[DEV-5822] Performance improvements

### PR DESCRIPTION
**Pull request by kbard**
**Description:**

Converted Postgres endpoints to count File D award ids instead of File C for performance.  Also tweaked the agency endpoint a bit further to squeak about even more performance.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation unaffected
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Materialized views unaffected
5. [x] Front end unaffected
6. [x] Data validation completed
7. [x] No Operations ticket required
8. [x] Jira Ticket [DEV-5822](https://federal-spending-transparency.atlassian.net/browse/DEV-5822):
    - [x] Link to this Pull-Request

### CODE REVIEW

File: usaspending_api/disaster/v2/views/award/count.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def post(self, request: Request) -> Response:
         count = FinancialAccountsByAwards.objects.filter(*filters)
         if self.award_type_codes:
             count = count.values("award_id").filter(award__type__in=self.filters.get("award_type_codes"))
-        count = self.when_non_zero_award_spending(count.annotate(unique_c=self.unique_file_c)).aggregate(
+        count = self.when_non_zero_award_spending(count.annotate(unique_c=self.unique_file_c_awards)).aggregate(
```
Comment (tony-sappe) [inline usaspending_api/disaster/v2/views/award/count.py:30]: this is getting confusing...

At first I though this needs to count file d, then I realized it should match award/amount.

We should just remove this endpoint

Comment (kbard) [reply usaspending_api/disaster/v2/views/award/count.py:30]: Preachin' to the choir...

File: usaspending_api/disaster/tests/integration/test_federal_account_total.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def test_federal_account_success(client, generic_account_data, monkeypatch, help
                 }
             ],
             "code": "000-0000",
-            "award_count": None,
```
Comment (tony-sappe) [inline usaspending_api/disaster/tests/integration/test_federal_account_total.py:28]: All responses for "total" spending should be returning None

Comment (kbard) [reply usaspending_api/disaster/tests/integration/test_federal_account_total.py:28]: Ah.  Ok.  Now your Slack comment makes more sense.  Got it.

**Pull request merged**